src/app/public/modules/adapter-service/adapter.service.ts

```diff
@@ -161,6 +161,14 @@ export class SkyCoreAdapterService {
     return elements;
   }
 
+  /**
+   * Returns the clientWidth of the provided elementRef.
+   * @param elementRef - The element to determine width.
+   */
+  public getWidth(elementRef: ElementRef): number {
```

Comment:
For all of these, should we just rake in the HTMLElement? There might be some places where we don’t have access to an ElementRef but still might need this functionality. Other functions in this adapter just take in an HTMLElement, though it appears we have been inconsistent with that.

Reply:
It's worth considering. I'm not sure what our pattern is, as I see it done differently across all our repos. Here's my take - The adapter services are Web API adapters and should offload that logic from the components. Since an `HTMLElement` is part of the Web API, the best case scenario would be to hand the adapter an `ElementRef` (angular specific), and then let the adapter suss out the `HTMLElement`. I know that's not possible in all scenarios, but we should shoot for it when we can.

Reply:
I get what you are saying. However, my concern is that we are placing this in `core` where we don't know what all the use cases for these methods are. To get an `ElementRef` you have to be passing in something that comes from a base component or that is created via the Renderer. I just worry that we are boxing ourselves in with when these methods can be used by having them take in `ElementRef` objects. I do agree that these are Web API adapters but I don't know that I agree that grabbing the `HTMLElement` has to be a part of what that adapter does. We are still abstracting the actual Web API methods and properties from being grabbed and we know that if we have an `ElementRef` we can get the `nativeElement`. We don't necessarily know that we will have the inverse to pass in. 

I'm not going to make a mountain out of a mole hill with it and will go ahead an approve :) I just don't think I 100% agree on this one :)